tests/validator20/validate_spec_test.py

```diff
@@ -423,3 +425,13 @@ def test_highlight_inconsistent_schema_object_validation(minimal_swagger_dict, s
     minimal_swagger_dict.update(swagger_dict_override)
     with pytest.raises(SwaggerValidationError):
         validate_spec(minimal_swagger_dict)
+
+
+def test_validation_performance(minimal_swagger_dict):
+    """Ensure repeat validations are much faster than the first"""
+
+    with mock.patch("swagger_spec_validator.common.read_file", wraps=read_file) as m:
+        for _ in range(3):
+            validate_spec(minimal_swagger_dict)
+
+    assert m.call_count <= 1
```

Comment:
I would strongly advice to have assertions that are as selective as possible.
In your scenario the function will be called exactly once.
Additionally to make it more specific you might also want to ensure that is invoked with the parameter that you would expect
```python
m.assert_called_once_with(resource_filename('swagger_spec_validator', 'schemas/v2.0/schema.json'))
```

Side notes:
 * a similar test should be added into validator12 tests
 * a unit-test for `read_resource_file` function would be added

Reply:
It's asserted this way because if the entire test suite is run, then the file is already cached, and it will be called zero times.

 If just this test is run, it will be called once

Reply:
It wasn't obvious to me how to make a test for validator12.

Instead I removed the validator20 test since it's redundant with the now added unit test.

Reply:
It makes sense ... if you add only the unit-test then it will be fine.

To workaround the issue of "what if the test is executed after other tests that have already loaded the resource?" I can suggest two possible approaches:
1. ensure that you clear the cache when you run the test (`read_resource_file.clear_cache()`)
2. fully mock `read_file` method and load a resource that you're sure that does not exist (so you'll be the only caller)

Reply:
I went with option 2 - thanks